features/deep-linking/vanity-url-handler.ts

```diff
@@ -0,0 +1,96 @@
+import { config } from "@/config"
+import { navigateWithReset } from "@/navigation/navigation.utils"
+import { deepLinkLogger } from "@/utils/logger/logger"
+import { findInboxIdByUsername } from "@/features/profiles/utils/find-inbox-id-by-username"
+
+/**
+ * Checks if a URL matches a vanity profile URL pattern
+ * Pattern 1: username.convos.org (subdomain)
+ * Pattern 2: convos.org/username (path)
+ */
+export function extractUsernameFromVanityUrl(url: string): string | null {
+  try {
+    // Make sure we have a valid URL
+    if (!url) return null
+    
+    // Normalize URL to ensure it has a protocol
+    const normalizedUrl = url.startsWith("http") ? url : `https://${url}`
```

Comment:
let's create afunction in utils general.ts i feel this could be useful in lots of places